src/com/google/debugging/sourcemap/proto/mapping.proto

```diff
@@ -22,4 +22,13 @@ message OriginalMapping {
 
   // The original name of the identifier.
   optional string identifier = 4;
+  
+  enum RetrievalType {
```

Comment:
```
  enum Precision {
    UNKNOWN_PRECISION = 0,
    EXACT = 1,
    APPROXIMATE_LINE = 2,
  }
```

Proto bindings in C++ and Python dump all the enum elements in the surrounding namespace, so it's a good idea to qualify UNKNOWN in some way.

Reply:
I'm somewhat inexperienced with protobuffs, but I believe I understand your point. 

Does it make sense to qualify `EXACT` and `APPROXIMATE_LINE` with `_PRECISION` as well, or would the example you provided suffice?

Reply:
Those should be fine, but it's a judgement call. `UNKNOWN` is a particularly bad example because almost every enum has an `UNKNOWN = 0` entry, so conflicts are very likely for that name.

Reply:
Ah I see, that makes sense.